### ISSUE TITLE
New module: gateleen-cache

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#399]
A new gateleen module should be introduced called gateleen-cache.

# Features needed
The new module should provide the following features:

- Cache responses from backends
- Caches should be stored in the redis storage
- Configuration resource to configure resources to cache
- Caches can expire
- [Optional] Mark caches to be held also in memory when heavily accessed. This omits the roundtrip to the storage

## Configuration resource example
The configuration resource could look something like this:
```json
{
    "/gateleen/the/resource/to/be/cached": {
        "target": "http://localhost:1234/resources/someResource,
        "expireAfter": 172800,
        "keepInMemory": true
    }
}
```

**Comment by hiddenalpha** [comment]
Please keep [HTTP_Caching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#controlling_caching) in mind.

**Comment by mcweba** [comment]
Released in [v1.2.5](https://github.com/swisspush/gateleen/releases/tag/v1.2.5)
